src/chains/ethereum/transaction/src/base-transaction.ts

```diff
@@ -22,11 +22,15 @@ export const calculateIntrinsicGas = (
   common: Common
 ) => {
   const hardfork = common.hardfork() as Hardfork;
-  // Set the starting gas for the raw transaction
-  let gas = Params.TRANSACTION_GAS;
-  // if it doesn't have a "to" address this is a contract creation and it costs
-  // `TRANSACTION_CREATION` more gas.
-  if (!hasToAddress) gas += Params.TRANSACTION_CREATION;
+  let gas: bigint;
+  // set the starting gas for the transaction
+  if (hasToAddress) {
+    gas = Params.TRANSACTION_GAS;
+  } else {
+    // if it doesn't have a "to" address this is a contract creation and it costs
+    // `TRANSACTION_CREATION` gas.
```

Comment:
```suggestion
    // `TRANSACTION_CREATION_GAS`.
```

Reply:
Additionally, I wonder if this is easier to read as a conditional:
```
let gas = hasToAddress
  ? Params.TRANSACTION_GAS
  : Params.TRANSACTION_CREATION_GAS;
```

🚴🏚

Reply:
but where would the comments go, hmmmmmm? :-D